src/storage/src/hummock/sstable_store.rs

```diff
@@ -509,24 +488,209 @@ impl SstableStoreWrite for SstableStore {
     }
 }
 
+pub struct BlockRange {
+    begin: usize,
+    end: usize,
+}
+
+/// Buffer SST data and upload it as a whole on `finish`.
+/// The upload is finished when the returned `JoinHandle` is joined.
+pub struct BatchUploadWriter {
+    sst_id: HummockSstableId,
+    sstable_store: SstableStoreRef,
+    policy: CachePolicy,
+    buf: BytesMut,
```

Comment:
We may store the raw bytes of each block instead of concatenating them into a single byte array so that when we later refill the block cache, we don't have to split the byte array into separate blocks. The `put_sst_data` of `SstableStore` and `put` of `ObjectStore` may now change to accept input as `Vec<Bytes>`.

Reply:
In terms of readability, wouldn't it be counter-intuitive that `Vec<Bytes>` represents a single object?

Reply:
It indeed looks wired. 

Shall we can introduce a new method named `put_sized_stream`, and then we pass a `Bytes` stream along with its total size?

Reply:
Then we will have 3 write interfaces in object store :lark-cry:. I think for this PR, we can just keep the implementation of batch upload the same as  before, and use another PR for this.

Reply:
Make sense.